engine/schema/src/main/java/com/cloud/upgrade/GuestOsMapper.java

```diff
@@ -74,18 +74,19 @@ private long getGuestOsIdFromHypervisorMapping(GuestOSHypervisorMapping mapping)
     }
 
     public void addGuestOsAndHypervisorMappings(long categoryId, String displayName, List<GuestOSHypervisorMapping> mappings) {
-        if (!addGuestOs(categoryId, displayName)) {
-            LOG.warn("Couldn't add the guest OS with category id: " + categoryId + " and display name: " + displayName);
-            return;
-        }
-
-        if (CollectionUtils.isEmpty(mappings)) {
-            return;
-        }
-
         long guestOsId = getGuestOsId(categoryId, displayName);
         if (guestOsId == 0) {
             LOG.debug("No guest OS found with category id: " + categoryId + " and display name: " + displayName);
+            if (!addGuestOs(categoryId, displayName)) {
+                LOG.warn("Couldn't add the guest OS with category id: " + categoryId + " and display name: " + displayName);
+                return;
+            }
+            guestOsId = getGuestOsId(categoryId, displayName);
+        } else {
+            // TODO: update is_user_defined to false
```

Comment:
In my opinion, we better leave the user_defined flag as is to keep track of operator activities.

Reply:
we can use the user_defined flag to decide which ones to remove and only keep the system defined one. In this case we, will have to allow for the ectra definition to begin with and deduplicate it right away.

Reply:
How about adding a new guest OS entry if the existing one is user-defined ?

We are anyways deduplicating the list from listGuestOS command starting 4.19 version, we will show only the latest entry if there duplicates. cc @DaanHoogland

Reply:
@harikrishna-patnala is there another place the deduplication is already taking place (see my pseudo code)?

I do agree we should look at the user defined flag during deduplication. seems like the cleanest way.

Reply:
@DaanHoogland I'm thinking of not doing any merge operation as it may affect the old entries in older versions as well.

I think the problem here is only that we have duplicate guest OS entries, which is not a critical/major issue. 
My proposal is to leave them as is and make fix for future entries.

Regarding the "Red Hat Enterprise Linux 9" and other similar entries, instead of doing a merge operation, I would suggest adding new mapping entries for both the names (RHEL and RHEL9*) so that if the operator uses any of the Guest OSs it won't behave differently while deploying VM (otherwise if the mapping is not found, CS may choose "default").

@weizhouapache @DaanHoogland